app/code/Magento/Catalog/Controller/Adminhtml/Product/AddAttributeToTemplate.php

```diff
@@ -75,33 +83,33 @@ class AddAttributeToTemplate extends \Magento\Catalog\Controller\Adminhtml\Produ
     /**
      * Constructor
      *
-     * @param \Magento\Backend\App\Action\Context $context
+     * @param Context $context
      * @param Builder $productBuilder
-     * @param \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
-     * @param \Magento\Eav\Api\Data\AttributeGroupInterfaceFactory|null $attributeGroupFactory
+     * @param JsonFactory $resultJsonFactory
+     * @param AttributeGroupInterfaceFactory|null $attributeGroupFactory
      * @SuppressWarnings(PHPMD.ExcessiveParameterList)
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Catalog\Controller\Adminhtml\Product\Builder $productBuilder,
-        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
-        \Magento\Eav\Api\Data\AttributeGroupInterfaceFactory $attributeGroupFactory = null
+        Context $context,
+        Builder $productBuilder,
+        JsonFactory $resultJsonFactory,
+        AttributeGroupInterfaceFactory $attributeGroupFactory = null
```

Comment:
The $attributeGroupFactory is longer than 20 characters, so it causes static tests to fail, however, we cannot change the variable name without bracking backward compatibility. In this exceptional situation there is nothing we can do, except adding a ` @SuppressWarnings(PHPMD.LongVariable)`

Reply:
Done, I will try to squash commits.